pkg/controllers/managementuser/rbac/roletemplates/roletemplate_handler.go

```diff
@@ -0,0 +1,232 @@
+}
+
+// Create func always creates:
+// 1. a ClusterRole with the same name as the RoleTemplate
+// 2. an Aggregating ClusterRole that aggregates all inherited RoleTemplates with the name "RoleTemplateName-aggregator"
+//
+// For RoleTemplates with the Context == "Project", we also create:
+// 1. If the RoleTemplate has any rules for Global Resources, make a ClusterRole with those named "RoleTemplateName-promoted"
+// 2. an Aggregating ClusterRole that aggregates all inherited RoleTemplates' promoted Cluster Roles named "RoleTemplateName-promoted-aggregator"
+func (rtl *roleTemplateLifecycle) Create(rt *v3.RoleTemplate) (runtime.Object, error) {
+	// 1. Create cluster role
+	cr := rtl.buildClusterRole(rt)
+	_, err := rtl.crClient.Create(cr)
+	if err != nil {
+		return nil, err
+	}
+	// 2. Create aggregating cluster role
+	cr = rtl.buildAggregatingClusterRole(rt)
+	_, err = rtl.crClient.Create(cr)
+	if err != nil {
+		return nil, err
+	}
+	// 3. Project global resources
+	if rt.Context == "project" {
+		// Create a promoted cluster role
+		cr = rtl.buildProjectPromotedClusterRole(rt)
+		_, err = rtl.crClient.Create(cr)
+		if err != nil {
+			return nil, err
+		}
+
+		// Create a promoted aggregating cluster role
+		cr = rtl.buildProjectPromotedAggregatedClusterRole(rt)
+		_, err = rtl.crClient.Create(cr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return rt, nil
+}
+
+func (rtl *roleTemplateLifecycle) Updated(rt *v3.RoleTemplate) (runtime.Object, error) {
+	// 1. Create cluster role
+	cr := rtl.buildClusterRole(rt)
+	_, err := rtl.crClient.Update(cr)
+	if err != nil {
+		return nil, err
+	}
+	// 2. Create aggregating cluster role
+	cr = rtl.buildAggregatingClusterRole(rt)
+	_, err = rtl.crClient.Update(cr)
+	if err != nil {
+		return nil, err
+	}
+	// 3. Project global resources
+	if rt.Context == "project" {
+		// Create a promoted cluster role
+		cr = rtl.buildProjectPromotedClusterRole(rt)
+		_, err = rtl.crClient.Update(cr)
+		if err != nil {
+			return nil, err
+		}
+
+		// Create a promoted aggregating cluster role
+		cr = rtl.buildProjectPromotedAggregatedClusterRole(rt)
+		_, err = rtl.crClient.Update(cr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return rt, nil
+}
+
+func (rtl *roleTemplateLifecycle) Remove(rt *v3.RoleTemplate) (runtime.Object, error) {
+	err := errors.Join(
+		rtl.crClient.Delete(rt.Name, &metav1.DeleteOptions{}),
+		rtl.crClient.Delete(rt.Name+"-aggregator", &metav1.DeleteOptions{}))
+	if rt.Context == "project" {
+		err = errors.Join(err,
+			rtl.crClient.Delete(rt.Name+"-promoted", &metav1.DeleteOptions{}),
+			rtl.crClient.Delete(rt.Name+"-promoted-aggregator", &metav1.DeleteOptions{}))
+	}
+	return nil, err
+}
+
+func (rtl *roleTemplateLifecycle) buildClusterRole(rt *v3.RoleTemplate) *v1.ClusterRole {
+	return &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: rt.Name,
+			// Label we use for aggregation
+			Labels: map[string]string{
+				rt.Name: "true",
```

Comment:
Can we make the produced labels more meaningful?
E.g. `aggregation.prefix/${rt.Name}=true`

Reply:
I've changed it to actually be of the form `management.cattle.io/aggregates: clusterRoleName` because I think that's more clear what's happening